stix_shifter_modules/arcsight/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,336 @@
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators, StartStopQualifier
+from stix_shifter_utils.stix_translation.src.patterns.errors import SearchFeatureNotSupportedError
+from datetime import datetime, timedelta
+import re
+import json
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+# timestamp format check for data source
+ISO_8601_PATTERN = r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([' \
+                   r'0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$ '
+MAC = r'(([0-9a-fA-F]{0,2}[:-]){1,5}([0-9a-fA-F]{0,2}))'
+IP_ADDRESS = r'((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){1,3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)'
+
+
+# update this lookup for long-int type fields in arcsight logger
+INTEGER_LOOKUP_FIELDS = ["sourcePort", "destinationPort", "priority", "startTime", "endTime", "baseEventCount",
+                         "eventId", "sourceMacAddress", "destinationMacAddress"]
+
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
+        ComparisonExpressionOperators.And: "AND",
+        ComparisonExpressionOperators.Or: "OR",
+        ComparisonComparators.GreaterThan: ">",
+        ComparisonComparators.GreaterThanOrEqual: ">=",
+        ComparisonComparators.LessThan: "<",
+        ComparisonComparators.LessThanOrEqual: "<=",
+        ComparisonComparators.Equal: "=",
+        ComparisonComparators.NotEqual: "!=",
+        ComparisonComparators.Like: "=",
```

Comment:
is there no equivalant  `Like` operator ? Equal doesn't seem right.

Reply:
LIKE keyword is not supported in ArcSight, however, the alternative for LIKE operator is the combination of "=" (Equals) and "*" asterisk operator (Equivalent to % in SQL LIKE).

Example: The pattern, process:name LIKE '%exe' is translated as, sourceProcessName = "*exe".

For the above pattern, the operator is assigned to "=" (Equals) and then the '%' is formatted to '*' during the query construction.